shells/webextension/src/ChromeStorageAdapter.js

```diff
@@ -0,0 +1,83 @@
+// Expects a chrome.storage
+// [`StorageArea`](https://developer.chrome.com/apps/storage#type-StorageArea)
+// input and exports a
+// [Storage](https://developer.mozilla.org/en-US/docs/Web/API/Storage) interface
+class ChromeStorageAdapter {
+  constructor(chromeStorage, readyCallback) {
+    if (!readyCallback) {
+      throw new TypeError(
+        "ChromeStorageAdapter called without a `readyCallback`",
+      );
+    }
+    this.chromeStorage = chromeStorage;
+
+    chromeStorage.get(
+      { apolloClientDevtools: {} },
+      ({ apolloClientDevtools }) => {
+        this.localCache = apolloClientDevtools;
+
+        readyCallback();
+      },
+    );
+  }
+
+  remoteWritePending = false;
+  needsRemoteWrite = false;
+  localCache = {};
+
+  updateRemoteStorage = () => {
+    if (this.remoteWritePending) {
+      this.needsRemoteWrite = true;
+
+      return;
+    }
+
+    this.remoteWritePending = true;
+    this.needsRemoteWrite = false;
+
+    this.chromeStorage.set({ apolloClientDevtools: this.localCache }, () => {
+      this.remoteWritePending = false;
+
+      if (this.needsRemoteWrite) {
+        this.updateRemoteStorage();
+      }
+    });
```

Comment:
quite thorough!

Reply:
I forgot to document this function. I'm going to do that before merging.